### PR TITLE
Add Client for Canvas REST API

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
1.  Add a RestClient for [access](https://canvas.instructure.com/doc/api/) to the Canvas LMS, in contrast to the [GraphQL](https://canvas.instructure.com/doc/api/file.graphql.html) api.
2. Add a teacher_classrooms endpoint for the RestClient

## WHY
1.  The client gives access to much of a particular canvas Instance's data.
2. This particular endpoint is needed for Classroom rostering

## HOW
1. Add the lms-api Gem which is a wrapper to the REST API.  
2. Pull data from courses data from courses endpoint and then iterate over each course to find each section and that sections students.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

### CODE REVIEW

File: services/QuillLMS/spec/factories/auth_credentials.rb [hunk 1]
```diff
@@ -30,7 +30,20 @@
     expires_at 1.day.from_now
     user
 
-    factory :canvas_auth_credential, parent: :auth_credential, class: :CanvasAuthCredential
+    factory :canvas_auth_credential, parent: :auth_credential, class: :CanvasAuthCredential do
```
Comment (brendanshean) [inline services/QuillLMS/spec/factories/auth_credentials.rb:33]: This factory's complexity is related to the associated `canvas_instance_auth_credential` record, which in practice should always be created after a `canvas_auth_credential` creation.  However, this introduces [circularity](https://github.com/empirical-org/Empirical-Core/pull/10713/files#diff-db6f061c3f4219b39d595eccb79944da776260222746a15fd99940eb9d5b29e5R24) when unit testing `canvas_instance_auth_credential` hence the `canvas_auth_credential_without_canvas_instance_auth_credential` factory.  I'm not sure of a cleaner way to do this as factory bot doesn't seem to allow for conditional callbacks and I don't want to introduce two new traits for default behavior.  I'm definitely open to suggestions here.

Comment (anathomical) [reply services/QuillLMS/spec/factories/auth_credentials.rb:33]: This is an awkward compromise, but I can't think of a cleaner way to do it offhand.

File: services/QuillLMS/spec/services/canvas_integration/classroom_data_adapter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    it { expect(subject).to eq classroom_data }
+  end
+
+  def classroom_data
+    {
+      name: classroom_name,
+      external_classroom_id: section_data[:id],
+      students: students
+    }
+  end
```
Comment (anathomical) [inline services/QuillLMS/spec/services/canvas_integration/classroom_data_adapter_spec.rb]: Thoughts on this just being a `let` call at the top?  The rest of the spec will look the same, but no one has to go look for this function definition.

**Review by anathomical — APPROVED** [review]
Question about spec organization, but this looks solid to me.

File: services/QuillLMS/spec/services/canvas_integration/classroom_data_adapter_spec.rb (hunk 2 above)
Comment (brendanshean) [reply services/QuillLMS/spec/services/canvas_integration/classroom_data_adapter_spec.rb]: Good idea!  Originally I had this as a function that passed parameters into but now I would agree that it should just be a let.

**Pull request merged**